src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -85,20 +86,24 @@ class Global(var currentSettings: Settings, reporter0: LegacyReporter)
 
   override def settings = currentSettings
 
-  private[this] var currentReporter: LegacyReporter = { reporter = reporter0 ; currentReporter }
```

Comment:
I guess you didn't like this trick. BTW, this usage of `locally` is what inspired the recent PR. I just took this up again, and sbt tells me that this is where I left off...

Reply:
> you didn't like this trick

Not that I didn't like it, it made me smile